applications/SeismicApplication/custom_conditions/fiber_beam_column_section.hpp

```diff
@@ -0,0 +1,304 @@
+    ///@{
+
+    /**
+     * Assignment Operator
+     */
+    FiberBeamColumnSection& operator=(FiberBeamColumnSection const& rOther) = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * Initializes all the fibers and initializes the local flexibility matrix.
+     */
+    void Initialize();
+
+    /**
+     * This method is to get the global flexibility matrix
+     * @param rGlobalFlexibilityMatrix: The global flexibility matrix
+     */
+    void GetGlobalFlexibilityMatrix(Matrix& rGlobalFlexibilityMatrix);
+
+    /**
+     * This method is to get the global deformation residuals
+     * @param rGlobalResiduals: The global deformation residuals
+     */
+    void GetGlobalDeformationResiduals(Vector& rGlobalResiduals);
+
+    /**
+     * This method updates the forces, deformations (and fiber strains).
+     * Then, updates the local flexibililty matrix, calculates the unbalance forces,
+     * and checks for convergence.
+     * @param rElementForceIncrements: the force increments of the element
+     * @return True if norm of the unbalance force is below the tolerance
+     */
+    bool StateDetermination(const Vector& rElementForceIncrements);
+
+    /**
+     * This method resets the deformation residuals to zero.
+     */
+    void ResetResidual();
+
+    /**
+     * Called when the non linear iterations have converged.
+     */
+    void FinalizeSolutionStep();
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    /// number of fibers
+    SizeType Size() { return mFibers.size(); }
+
+    /// numerical integration weight
+    double GetWeight() { return mWeight; }
+
+    /// set the fibers of the section
+    void SetFibers (std::vector<FiberBeamColumnUniaxialFiber> Fibers) { mFibers = std::move(Fibers); }
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const;
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+
+private:
+
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    IndexType mId;       // id of the section
+    double mPosition;    // position of the integration point
+    double mWeight;      // weight of the integration point
+    double mTolerance;   // tolerance of the section equilibrium
```

Comment:
This should be aproeprty, saving in each section is super expensive

Reply:
This in the other hand should be defined in a shared object to reduce memory needs

Reply:
It is used in one function. I could simply pass it to the function from the element without saving it.

Reply:
For example